kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveGlobalConsumer.java

```diff
@@ -0,0 +1,398 @@
+ *   instead of only the requested partition. This is necessary because of the way that
+ *   the {@link org.apache.kafka.streams.processor.internals.GlobalStateManagerImpl}
+ *   handles restore. Specifically, it will loop partition-by-partition and update
+ *   the store for those partitions. The issue with this is that if we use {@code subscribe}
+ *   in place of {@code assign} we may get events from partitions that were not
+ *   specified in the {@code assign} call. These records will be dropped.</li>
+ *
+ *   <li>It commits before every call to {@link #poll(Duration)} because
+ *   otherwise the Streams implementation for global stores will never commit. We
+ *   should consider changing this to every {@code N} polls, but this would be
+ *   more difficult to test (we could make it configurable).</li>
+ * </ol>
+ *
+ * <p>This class breaks a lot of abstraction barriers, but allows us to
+ * support remote {@link org.apache.kafka.streams.kstream.GlobalKTable}s without
+ * forking Kafka Streams.</p>
+ */
+public class ResponsiveGlobalConsumer implements Consumer<byte[], byte[]> {
+
+  private final Consumer<byte[], byte[]> delegate;
+  private final Integer defaultApiTimeoutMs;
+  private final Admin admin;
+
+  public ResponsiveGlobalConsumer(
+      final Map<String, Object> config,
+      final Consumer<byte[], byte[]> delegate,
+      final Admin admin
+  ) {
+    final ConsumerConfig consumerConfig = new ConsumerConfig(config);
+
+    this.delegate = delegate;
+    this.defaultApiTimeoutMs = consumerConfig.getInt(ConsumerConfig.DEFAULT_API_TIMEOUT_MS_CONFIG);
+    this.admin = admin;
+  }
+
+  @Override
+  public Set<TopicPartition> assignment() {
+    return delegate.assignment();
+  }
+
+  @Override
+  public Set<String> subscription() {
+    return delegate.subscription();
+  }
+
+  @Override
+  public void subscribe(final Collection<String> topics, final ConsumerRebalanceListener listener) {
+    delegate.subscribe(topics, listener);
+  }
+
+  @Override
+  public void subscribe(final Collection<String> topics) {
+    delegate.subscribe(topics);
+  }
+
+  @Override
+  public void subscribe(final Pattern pattern, final ConsumerRebalanceListener listener) {
+    delegate.subscribe(pattern, listener);
+  }
+
+  @Override
+  public void subscribe(final Pattern pattern) {
+    delegate.subscribe(pattern);
+  }
+
+  @Override
+  public void unsubscribe() {
+    delegate.unsubscribe();
+  }
+
+  @Override
+  public void assign(final Collection<TopicPartition> partitions) {
+    subscribe(
+        partitions
+            .stream()
+            .map(TopicPartition::topic)
+            .collect(Collectors.toSet())
+    );
+  }
+
+  @Override
+  @Deprecated
+  public ConsumerRecords<byte[], byte[]> poll(final long timeoutMs) {
+    return poll(Duration.ofMillis(timeoutMs));
+  }
+
+  @Override
+  public ConsumerRecords<byte[], byte[]> poll(final Duration timeout) {
+    commitSync();
```

Comment:
Man, yikes...at the very least, if we're going to do this in the POC, let's make sure to maximize the number of records we can process each time. 100 is the default `max.poll.records` which is way too low, I'd say 1,000 or 10,000 even 

10,000 could be significant memory pressure though, and unfortunately last I checked (admittedly several years ago) the consumer's size-related configs don't provide a way to limit the fetch buffer memory completely. WDYT about 1,000?